### PR TITLE
First pass on using Display KHR, needs proper init flags

### PR DESCRIPTION
**Pull request by amverweij**
All credits go to Conrad Kreyling. When combined with sufficiently new mesa, this fixes issue #13732.

### CODE REVIEW

**Comment by konistehrad** [comment]
Thank you for opening this I’m currently not available to spearhead or rebase this but I can try to clarify any questions related to this PR.

**Comment by amverweij** [comment]
@konistehrad : thanks for your response! 
I forked your repo so that I can follow up on remarks and/or maintenance myself. Let's see how things go.

Actually , I wondered what you meant by "needs proper init flags". The code looks fine to me.

**Comment by hrydgard** [comment]
I'm okay to merge this, but is it really working? From the discussions on Discord I saw about this, it seemed more is needed...

File: Common/GPU/Vulkan/VulkanContext.cpp [hunk 1]
```diff
@@ -795,6 +800,15 @@ VkResult VulkanContext::ReinitSurface() {
 		break;
 	}
 #endif
+#if defined(VK_USE_PLATFORM_DISPLAY_KHR)
+	case WINDOWSYSTEM_DISPLAY:
+	{
+		VkDisplaySurfaceCreateInfoKHR display{ VK_STRUCTURE_TYPE_DISPLAY_SURFACE_CREATE_INFO_KHR };
+		display.flags = 0;
```
Comment (konistehrad) [inline Common/GPU/Vulkan/VulkanContext.cpp:807]: I was specifically concerned about which flags to pass Retroarch here. It seems that `0` works, though.

**Pull request merged**